core/services/cable_guy/main.py

```diff
@@ -86,22 +80,16 @@ def configure_interface(interface: EthernetInterface = Body(...)) -> Any:
 @version(1, 0)
 def add_address(interface_name: str, mode: AddressMode, ip_address: Optional[str] = None) -> Any:
     """REST API endpoint to add an IP address to an ethernet interface."""
-    try:
-        manager.add_ip(interface_name, mode, ip_address)
-        manager.save()
-    except Exception as error:
-        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(error)) from error
+    manager.add_ip(interface_name, mode, ip_address)
+    manager.save()
```

Comment:
This was returning bad request and not internal error

Reply:
It was because we didn't have a convention. Bad requests (400) should be returned only if we know that this error is caused by a faulty usage from the client (which we cannot, in this case).